bin/millau/runtime/src/rialto.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Configuration parameters for the Rialto Substrate chain.
+
+use bp_rialto::Header;
+use hex_literal::hex;
+use pallet_substrate_bridge::AuthoritySet;
+use sp_core::crypto::Public;
+use sp_finality_grandpa::AuthorityId;
+use sp_std::vec;
+
+/// The first header known to the pallet.
+///
+/// Note that this does not need to be the genesis header of the Rialto
+/// chain since the pallet may start at any arbitrary header.
+// To get this we first need to call the `chain_getBlockHash` RPC method, and then
+// we can use the result from that and call the `chain_getBlock` RPC method to get
+// the rest of the info.
+//
+// In this case we've grabbed the genesis block of the Rialto Substrate chain.
+pub fn initial_header() -> Header {
+	Header {
+		parent_hash: Default::default(),
+		number: Default::default(),
+		state_root: hex!("8fc2e3a8216b9f70fe2bc6804febb3f5192cb4046db5d546d1ad499a29bbbf27").into(),
+		extrinsics_root: hex!("03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c111314").into(),
+		digest: Default::default(),
+	}
+}
+
+/// The first set of Grandpa authorities known to the pallet.
+///
+/// Note that this doesn't have to be the "genesis" authority set, as the
+/// pallet can be configured to start from any height.
+pub fn initial_authority_set() -> AuthoritySet {
+	let set_id = 0;
+	let authorities = vec![
+		(AuthorityId::from_slice(&[1; 32]), 1),
+		(AuthorityId::from_slice(&[2; 32]), 1),
+		(AuthorityId::from_slice(&[3; 32]), 1),
+	];
```

Comment:
I wanted to get the real authorities + weights from Rialto at genesis but wasn't able to do so. I know I need to get the storage values under `GRANDPA_AUTHORITIES_KEY` using the `state_getStorage` RPC call but I wasn't able to figure out how

Reply:
There are two alternatives:
1) manually craft that value - initial authorities are Alice, Bob, Charlie, Dave, Eve and weight is 1 for every authority;
2) we have `substrate_relay::Client::grandpa_authorities_set()` - you may call it manually, or just run regulat Rialto -> RialtoPoA headers relay && when PoA contract is deployed, relay is tracing hex-encoded authorities set.

Reply:
> manually craft that value - initial authorities are Alice, Bob, Charlie, Dave, Eve and weight is 1 for every authority

I didn't do this because I didn't know what the default weight was for authorities. I guess it is indeed `1` though.

Also, when looking at our Rialto deployment I saw that every time the authorities change only three are selected, not five. I wasn't sure if during genesis our authority list should only consist of the three authorities chosen at the time, or the "full list" which we've set up in our `docker-compose`.

Reply:
At genesis it is the full list, then (starting from session 2) only 3 -of-5 authorities are selected